tests/scripts/generate_tests_pk_can_do_psa.py

```diff
@@ -0,0 +1,366 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import re
+
+# PK contexts to be tested
+PK_CONTEXTS = {
+    'MBEDTLS_PK_ECDSA': {
+        'PSA_KEY_TYPE_ECC_KEY_PAIR(MBEDTLS_TEST_PSA_ECC_ONE_FAMILY):MBEDTLS_TEST_PSA_ECC_ONE_CURVE_BITS': {
+            'MBEDTLS_PK_ALG_ECDSA(PSA_ALG_ANY_HASH)': [
+                'PSA_KEY_USAGE_SIGN_HASH',
+                'PSA_KEY_USAGE_VERIFY_HASH',
+            ],
+            'MBEDTLS_PK_ALG_ECDSA(PSA_ALG_SHA_256)': [
+                'PSA_KEY_USAGE_SIGN_HASH',
+                'PSA_KEY_USAGE_VERIFY_HASH',
+            ],
+        },
+        'PSA_KEY_TYPE_ECC_PUBLIC_KEY(MBEDTLS_TEST_PSA_ECC_ONE_FAMILY):MBEDTLS_TEST_PSA_ECC_ONE_CURVE_BITS': {
+            'MBEDTLS_PK_ALG_ECDSA(PSA_ALG_ANY_HASH)': [
+                'PSA_KEY_USAGE_VERIFY_HASH',
+            ],
+            # Skip context with PSA_ALG_SHA_256 because psa_can_do_psa() cannot verify which alg + hash_alg
+            # is associated with public key.
+        },
+    },
+    'MBEDTLS_PK_ECKEY': {
+        'PSA_KEY_TYPE_ECC_KEY_PAIR(MBEDTLS_TEST_PSA_ECC_ONE_FAMILY):MBEDTLS_TEST_PSA_ECC_ONE_CURVE_BITS': {
+            'MBEDTLS_PK_ALG_ECDSA(PSA_ALG_ANY_HASH)': [
+                'PSA_KEY_USAGE_SIGN_HASH',
+                'PSA_KEY_USAGE_VERIFY_HASH',
+            ],
+            'MBEDTLS_PK_ALG_ECDSA(PSA_ALG_SHA_256)': [
+                'PSA_KEY_USAGE_SIGN_HASH',
+                'PSA_KEY_USAGE_VERIFY_HASH',
+            ],
+            'PSA_ALG_ECDH': [
+                'PSA_KEY_USAGE_DERIVE',
+                # 'PSA_KEY_USAGE_DERIVE_PUBLIC',
+            ],
+        },
+        'PSA_KEY_TYPE_ECC_PUBLIC_KEY(MBEDTLS_TEST_PSA_ECC_ONE_FAMILY):MBEDTLS_TEST_PSA_ECC_ONE_CURVE_BITS': {
+            'MBEDTLS_PK_ALG_ECDSA(PSA_ALG_ANY_HASH)': [
+                'PSA_KEY_USAGE_VERIFY_HASH',
+            ],
+            # Skip context with PSA_ALG_SHA_256 because psa_can_do_psa() cannot verify which alg + hash_alg
+            # is associated with public key.
+
+            # 'PSA_ALG_ECDH': [
```

Comment:
A public key should say yes for ECDH with `DERIVE_PUBLIC`, that was the whole point of adding `DERIVE_PUBLIC`.

Reply:
Yeah, all the `DERIVE_PUBLIC` tests are commented out because I had issues in creating the PK context to be tested. I need to figure out how to do that.